### PR TITLE
Compatible with cpp17

### PR DESCRIPTION
**Pull request by lychees**
Make it easier for multiplayer version to include some cpp17 headers.
No side effects.

### CODE REVIEW

**Comment by Ghabry** [comment]
Jenkins: Test this please <3

File: src/pending_message.cpp [hunk 1]
```diff
@@ -132,7 +132,7 @@ std::string PendingMessage::ApplyTextInsertingCommands(std::string input, uint32
 		// Fast path - no substitutions occured, so just move the input into the return value.
 		output = std::move(input);
 	} else {
-		output.append(start_copy, end);
+		output.append(start_copy, end - start_copy);
```
Comment (Ghabry) [inline src/pending_message.cpp:135]: No side effects but doesn't this substraction change the API?

Comment (lychees) [reply src/pending_message.cpp:135]: No, it is same.
https://www.cplusplus.com/reference/string/string/append/

Comment (Ghabry) [reply src/pending_message.cpp:135]: Verified this now also in my IDE: This picks a different overload... C++ is a mess :)

**Review by Ghabry — APPROVED** [review]

**Review by fdelapena — APPROVED** [review]

**Pull request merged**